kedro/utils.py

```diff
@@ -78,3 +80,18 @@ def _find_kedro_project(current_dir: Path) -> Any:  # pragma: no cover
         if _is_project(parent_dir):
             return parent_dir
     return None
+
+
+@lru_cache(maxsize=None)
+def _has_rich_handler(logger: logging.Logger) -> bool:
+    """Returns true if the logger has a RichHandler attached."""
+    try:
+        from rich.logging import RichHandler
```

Comment:
why do we need to explicitly import `rich` for testing?

Reply:
My idea was checking if rich is installed at all, because if it isn't, the handler is not going to be there, making the function automatically return false. If it is installed, the user might still not be using it on their logging config, so we move to checking if it's present.

This approach is not working though, I'm verifying it right now.